lib/mixins/listen-on-root.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Issue #569: collapse::toggle::state triggered too many times
+ * @link https://github.com/bootstrap-vue/bootstrap-vue/issues/569
+ */
+export default {
+    data() {
```

Comment:
I'm not a fan of [this method for non-reactive props](https://github.com/vuejs/vue/issues/1988) because it has the potential to overwrite data.